### PR TITLE
Ignore more Paypal IPN transaction types

### PR DESCRIPTION
**Pull request by notbakaneko**
Or maybe should ignore everything without `invoice` field set 🤔 (which I recall wasn't done to make sure that callbacks for outgoing transactions that had them missing would explode)

### CODE REVIEW

File: app/Libraries/Payments/PaypalPaymentProcessor.php [hunk 1]
```diff
@@ -70,16 +70,16 @@ public function getNotificationType(): string
         static $rejectedStatuses = ['Declined', 'Denied', 'Expired', 'Failed', 'Voided'];
 
         $status = $this->getNotificationTypeRaw();
-        if (in_array($status, $paymentStatuses, true)) {
+        if ($this->shouldIgnore($status)) {
```
Comment (nanaya) [inline app/Libraries/Payments/PaypalPaymentProcessor.php:73]: probably can use a note on why this needs to be first (or maybe just inline the function)

File: app/Libraries/Payments/PaypalPaymentProcessor.php [hunk 2]
```diff
@@ -70,16 +70,16 @@ public function getNotificationType(): string
...
+            return NotificationType::IGNORED;
+        } elseif (in_array($status, $paymentStatuses, true)) {
```
Comment (nanaya) [inline app/Libraries/Payments/PaypalPaymentProcessor.php:75]: semi unrelated but this looks like a job for switch/match.

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Libraries/Payments/PaypalPaymentProcessor.php (hunk 2 above)
Comment (notbakaneko) [reply app/Libraries/Payments/PaypalPaymentProcessor.php:77]: I don't think it's worth switching it right now

**Review by nanaya — APPROVED** [review]
the comment doesn't really make sense without knowing how `getNotificationTypeRaw` works but oh well ಠ_ಠ

(the abstraction itself doesn't seem to be useful anymore either)

**Pull request merged**